src/tests/ftest/control/dmg_pool_query_test.py

```diff
@@ -0,0 +1,171 @@
+
+        # Clean up empty string from each list. 'pool_query' should look like:
+        # [['217afbae-534f-4d8f-9e49-74cbfb8ad155', '8', '0'],
+        # ['8'],
+        # ['2.0 GB', '2.0 GB', '250 MB', '250 MB', '250 MB'],
+        # ['16 GB', '16 GB', '2.0 GB', '2.0 GB', '2.0 GB'],
+        # ['0', '0']]
+        pool_query_info = {}
+        if pool_query:
+            for idx, info in enumerate(pool_query):
+                pool_query[idx] = [i for i in info if i]
+
+            pool_query_info = {
+                "pool_info": pool_query[0],
+                "target_cnt": pool_query[1],
+                "scm_info": pool_query[2],
+                "nvme_info": pool_query[3],
+                "rebuild_info": pool_query[4]
+            }
+        return pool_query_info
+
+    def test_pool_query_basic(self):
+        """
+        JIRA ID: DAOS-2976
+
+        Test Description: Test basic dmg functionality to query pool info on
+        the system. Provided a valid pool UUID, verify the output received from
+        pool query command.
+
+        :avocado: tags=all,small,pr,hw,dmg,pool_query,basic,poolquerybasic
+        """
+        self.log.info("==>   Verify dmg output against expected output:")
+        dmg_info = self.get_pool_query_info(self.uuid)
+        exp_info = {
+            "pool_info": self.params.get("pool_info", "/run/exp_vals/"),
+            "target_cnt": self.params.get("target_cnt", "/run/exp_vals/"),
+            "scm_info": self.params.get("scm_info", "/run/exp_vals/"),
+            "nvme_info": self.params.get("nvme_info", "/run/exp_vals/"),
+            "rebuild_info": self.params.get("rebuild_info", "/run/exp_vals/"),
+        }
+
+        # Add the expected uuid value
+        exp_info["pool_info"].insert(0, self.uuid.upper())
+
+        if exp_info != dmg_info:
+            self.log.info("==>   Found difference in dmg output and expected.")
+            self.fail("dmg pool-query: \n{} \nexpected: \n{}".format(
+                dmg_info, exp_info))
+        else:
+            self.log.info("==>   Expect values found in dmg pool query output.")
+
+    def test_pool_query_inputs(self):
+        """
+        JIRA ID: DAOS-2976
+
+        Test Description: Test basic dmg functionality to query pool info on
+        the system. Verify the inputs that can be provided to 'query --pool'
+        argument of the dmg pool subcommand.
+
+        :avocado: tags=all,small,pr,hw,dmg,pool_query,basic,poolqueryinputs
+        """
+        # Get test UUIDs
+        errors_list = []
+        uuids = self.params.get("uuids", '/run/pool_uuids/*')
+
+        # Disable raising an exception if the dmg command fails
+        self.dmg.exit_status_exception = False
+
+        for uuid in uuids:
+            self.log.info("\n==>   Using test UUID: %s", uuid[0])
+            self.log.info("==>   Test is expected to finish with: %s", uuid[1])
+
+            # Verify
+            if self.get_pool_query_info(uuid[0]):
```

Comment:
Isn't it better to save the output of get_pool_query_info and use it for if and elif lines so that we don't potentially call dmg pool query twice?

Reply:
Ok, will update.